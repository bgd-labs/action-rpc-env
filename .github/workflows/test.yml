# creates a new npm release
name: test

on:
  pull_request:
  push:
    branches:
      - main

jobs:
  test:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@eef61447b9ff4aafe5dcd4e0bbf5d482be7e7871 # pin@v4.2.1

      - uses: bgd-labs/github-workflows/.github/actions/setup-node@feat/actions-setup-node

      - name: test lib
        run: npm test -- --run

      - name: test action
        uses: ./
        with:
          ALCHEMY_API_KEY: "bla"
          RPC_POLYGON: "https://rpc.polygon.com"
      - name: test action (assertion)
        run: |
          test "${{env.RPC_POLYGON}}" = "https://rpc.polygon.com"
          test "${{env.RPC_MAINNET}}" = "https://eth-mainnet.g.alchemy.com/v2/bla"
