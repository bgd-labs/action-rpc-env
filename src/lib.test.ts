import { describe } from "node:test";
import { expect, it } from "vitest";
import { getRPCUrl } from "./lib";

describe("lib", () => {
	it("should use env var if given", () => {
		process.env.RPC_MAINNET = "https://rpc.mainnet.com";
		expect(getRPCUrl(1)).toMatchInlineSnapshot(`"https://rpc.mainnet.com"`);
	});

	it("should return undefined if no env var is given and alchemy key not passed", () => {
		process.env.RPC_MAINNET = "";
		expect(getRPCUrl(1)).toBeUndefined();
	});

	it("should generate url if no env var is given and alchemy key is passed", () => {
		process.env.RPC_MAINNET = "";
		expect(getRPCUrl(1, "abc")).toMatchInlineSnapshot(
			`"https://eth-mainnet.g.alchemy.com/v2/abc"`,
		);
	});
});
