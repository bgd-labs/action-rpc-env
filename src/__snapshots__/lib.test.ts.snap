// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should return alchemy supported chain 1 > eth-mainnet 1`] = `"https://eth-mainnet.g.alchemy.com/v2/abc"`;

exports[`should return alchemy supported chain 10 > opt-mainnet 1`] = `"https://opt-mainnet.g.alchemy.com/v2/abc"`;

exports[`should return alchemy supported chain 56 > bnb-mainnet 1`] = `"https://bnb-mainnet.g.alchemy.com/v2/abc"`;

exports[`should return alchemy supported chain 100 > gnosis-mainnet 1`] = `"https://gnosis-mainnet.g.alchemy.com/v2/abc"`;

exports[`should return alchemy supported chain 137 > polygon-mainnet 1`] = `"https://polygon-mainnet.g.alchemy.com/v2/abc"`;

exports[`should return alchemy supported chain 250 > fantom-mainnet 1`] = `"https://fantom-mainnet.g.alchemy.com/v2/abc"`;

exports[`should return alchemy supported chain 324 > zksync-mainnet 1`] = `"https://zksync-mainnet.g.alchemy.com/v2/abc"`;

exports[`should return alchemy supported chain 1088 > metis-mainnet 1`] = `"https://metis-mainnet.g.alchemy.com/v2/abc"`;

exports[`should return alchemy supported chain 1101 > polygonzkevm-mainnet 1`] = `"https://polygonzkevm-mainnet.g.alchemy.com/v2/abc"`;

exports[`should return alchemy supported chain 4002 > fantom-testnet 1`] = `"https://fantom-testnet.g.alchemy.com/v2/abc"`;

exports[`should return alchemy supported chain 8453 > base-mainnet 1`] = `"https://base-mainnet.g.alchemy.com/v2/abc"`;

exports[`should return alchemy supported chain 42161 > arb-mainnet 1`] = `"https://arb-mainnet.g.alchemy.com/v2/abc"`;

exports[`should return alchemy supported chain 43113 > avax-fuji 1`] = `"https://avax-fuji.g.alchemy.com/v2/abc"`;

exports[`should return alchemy supported chain 43114 > avax-mainnet 1`] = `"https://avax-mainnet.g.alchemy.com/v2/abc"`;

exports[`should return alchemy supported chain 80002 > polygon-amoy 1`] = `"https://polygon-amoy.g.alchemy.com/v2/abc"`;

exports[`should return alchemy supported chain 84532 > base-sepolia 1`] = `"https://base-sepolia.g.alchemy.com/v2/abc"`;

exports[`should return alchemy supported chain 421614 > arb-sepolia 1`] = `"https://arb-sepolia.g.alchemy.com/v2/abc"`;

exports[`should return alchemy supported chain 534351 > scroll-sepolia 1`] = `"https://scroll-sepolia.g.alchemy.com/v2/abc"`;

exports[`should return alchemy supported chain 534352 > scroll-mainnet 1`] = `"https://scroll-mainnet.g.alchemy.com/v2/abc"`;

exports[`should return alchemy supported chain 11155111 > eth-sepolia 1`] = `"https://eth-sepolia.g.alchemy.com/v2/abc"`;

exports[`should return alchemy supported chain 11155420 > opt-sepolia 1`] = `"https://opt-sepolia.g.alchemy.com/v2/abc"`;
